### PR TITLE
feat(bucketeer): implement variation methods

### PR DESCRIPTION
**Pull request by mura-s**

### CODE REVIEW

File: pkg/bucketeer/sdk.go [hunk 1]
```diff
@@ -1,87 +1,256 @@
...
+	if err != nil || res == nil {
+		if status.Code(err) == codes.DeadlineExceeded {
+			s.eventProcessor.PushTimeoutErrorCountMetricsEvent(ctx, s.tag)
+		} else {
+			s.eventProcessor.PushInternalErrorCountMetricsEvent(ctx, s.tag)
+		}
+		return nil, fmt.Errorf("failed to get evaluation: %w", err)
+	}
+	s.eventProcessor.PushGetEvaluationLatencyMetricsEvent(ctx, time.Since(reqStart), s.tag, res.State.String())
+	s.eventProcessor.PushGetEvaluationSizeMetricsEvent(ctx, proto.Size(res), s.tag, res.State.String())
```
Comment (cre8ivejp) [inline pkg/bucketeer/sdk.go:211]: To speed up the response, we could send these events `PushGetEvaluationLatencyMetricsEvent
` and `PushGetEvaluationSizeMetricsEvent` asynchronously like we do in the client SDK. 
What do you think?

File: pkg/bucketeer/event/processor.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func (p *processor) PushEvaluationEvent(
+	ctx context.Context,
+	user *protouser.User,
+	evaluation *protofeature.Evaluation,
+) {
+}
+
+func (p *processor) PushDefaultEvaluationEvent(ctx context.Context, user *protouser.User, featureID string) {
```
Comment (cre8ivejp) [inline pkg/bucketeer/event/processor.go:69]: Don't we need to pass the default value to be reported as well?

Comment (mura-s) [reply pkg/bucketeer/event/processor.go:69]: No, EvaluationEvent doesn't have a default value field.

https://github.com/ca-dp/bucketeer-android-sdk/blob/master/bucketeer/src/main/java/jp/bucketeer/sdk/events/EventCreator.kt#L45-L60

File: pkg/bucketeer/sdk.go (hunk 1 above)
Comment (mura-s) [reply pkg/bucketeer/sdk.go:211]: I think the speed is no problem because eventProcessor pushes these events to the in-memory queue.

Comment (cre8ivejp) [reply pkg/bucketeer/sdk.go:211]: Ok, let's see how it goes when you run the tests.

**Review by cre8ivejp — APPROVED** [review]
👍

**Pull request merged**